### PR TITLE
Rework channel points dynamic pricing: unify cooldowns, time-based half-life, derived increment

### PR DESCRIPTION
**Pull request by ExpiredMinotaur**
## Summary

Three changes to dynamic channel points pricing, as requested:

- **Unify the two cooldown settings.** A reward's dynamic-pricing "Cooldown (s)" field and its actual Twitch "Global cooldown (s)" could previously drift apart. Pricing's `cooldown_seconds` is no longer a separately-edited form field — it's always derived from the reward's live Twitch global cooldown (or a 60s fallback when the reward has none), synced both when pricing is saved and when the reward itself is edited.
- **Half-life is now a fixed duration, per streamer.** Previously `decay_half_life_periods` was a bot-wide setting expressed in "cooldown periods" (so effective decay time varied per reward). It's now `half_life_seconds`, a plain time duration configured per streamer (default 30 minutes), replacing the old singleton `pricing_global_settings` table with a per-streamer `reward_pricing_settings` table.
- **Redemption increment is now derived, not configured.** Instead of a flat global `redemption_increment`, each redemption's demand bump is computed as `cooldown_seconds / (time_to_max_multiplier * half_life_seconds)` — i.e. redeeming at a reward's own max frequency reaches 100% demand after `time_to_max_multiplier` half-lives (defaulting to 2x, so a 10-minute-cooldown reward with the 30-minute default half-life adds 1/6 demand per redemption).

## Details

- `src/twitch/pricing/rewardPricingMath.ts`: `decayDemand` now takes a fixed `halfLifeSeconds` instead of `cooldownSeconds` + `decayHalfLifePeriods`; added `computeRedemptionIncrement`; removed the unused `cooldownSeconds` field from `RewardPricingConfig` (it was never read by `computePrice`).
- `src/db/rewardPricing.ts`: replaced `getGlobalPricingSettings`/`saveGlobalPricingSettings`/`initGlobalPricingSettings` with per-streamer `getPricingSettingsForStreamer`/`savePricingSettingsForStreamer`; added `updatePricingCooldownForReward` to keep a pricing config's cooldown in sync with the reward's Twitch cooldown.
- `src/web/routes/channelPointsAdminMutations.ts`: the pricing-save route derives `cooldown_seconds` from a fresh Twitch lookup (falling back to the existing config, then the default); the reward-edit route re-syncs an existing pricing config's cooldown after every edit; the settings route (`POST /settings/pricing`, renamed from `/settings/global`) is now per-streamer instead of bot-owner-only.
- `views/channelPointsAdmin.ejs`: removed the editable "Cooldown (s)" input from the per-reward pricing form (now a read-only synced value); replaced the owner-only "Global Pricing Settings" section with a per-streamer "Pricing Settings" section (half-life in minutes, time-to-max multiplier).
- `migrations/reward_pricing_settings.sql` / `schema.sql` / `DATABASE-SCHEMA.md`: drop `pricing_global_settings`, add `reward_pricing_settings` (per-streamer).

## Test plan

- [x] `npx tsc --noEmit` passes
- [x] `npm test` passes (2052 tests)
- [ ] Manually verify in the Channel Points admin UI: create a reward with a Twitch global cooldown, enable dynamic pricing, confirm the pricing "Cooldown (s)" display matches; edit the reward's Twitch cooldown and confirm the pricing config updates to match; adjust the per-streamer half-life/time-to-max settings and confirm demand decay/increment behavior over a few redemptions matches expectations.

---
_Generated by [Claude Code](https://claude.ai/code/session_01V2KQajhJAQAdoe4XGFBT6e)_

## Summary by CodeRabbit

* **New Features**
  * Added streamer-specific Channel Points dynamic pricing settings (demand half-life and time-to-max-demand multiplier).
  * Added a reward pricing sync that automatically aligns pricing cooldown with the reward’s Twitch cooldown.

* **Bug Fixes**
  * Improved reward pricing update/delete flows to better tolerate Twitch cooldown lookup or sync failures while still completing the main action.

* **UI**
  * Updated the Channel Points admin page to replace global pricing settings with streamer-specific pricing settings.
  * Changed the per-reward “Cooldown (s)” field to read-only synced cooldown.

### CODE REVIEW

File: src/db/rewardPricing.ts [hunk 1]
```diff
@@ -200,44 +205,53 @@ export async function deletePricingConfig(id: number, streamerId: number): Promi
...
-     VALUES (1, ?, ?) AS new_row
+    `INSERT INTO reward_pricing_settings (streamer_id, half_life_seconds, time_to_max_multiplier)
+     VALUES (?, ?, ?) AS new_row
      ON DUPLICATE KEY UPDATE
-       decay_half_life_periods=new_row.decay_half_life_periods, redemption_increment=new_row.redemption_increment`,
-    [settings.decay_half_life_periods, settings.redemption_increment],
+       half_life_seconds=new_row.half_life_seconds, time_to_max_multiplier=new_row.time_to_max_multiplier`,
+    [streamerId, settings.half_life_seconds, settings.time_to_max_multiplier],
   );
 }
```
[the remark this answers is not in the document]
Comment (ExpiredMinotaur) [reply src/db/rewardPricing.ts:257]: Thanks for checking, but this is intentional and pre-existing, not something this PR introduces. The row-alias `VALUES (...) AS new_row ON DUPLICATE KEY UPDATE` form is the project's established, documented standard for every upsert in the codebase (see `CLAUDE.md`: "MySQL 8 upsert: Row-alias form only... Deprecated `VALUES(col)` not used"). `upsertPricingConfig` in this same file already uses this exact pattern, and the function I replaced (`saveGlobalPricingSettings`) used it too — I'm just following the existing convention for the new `savePricingSettingsForStreamer`. If the project needs to support MySQL <8.0.19 that'd be a repo-wide migration, out of scope here. Not changing this.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by ExpiredMinotaur** [comment]
Re: the `computeRedemptionIncrement` guard nitpick in `rewardPricingMath.ts` — confirming what was asked: yes, `half_life_seconds` and `time_to_max_multiplier` are already guaranteed positive both at the form layer (`parsePositiveNumberField` in `POST /settings/pricing`, plus the added `half_life_seconds < 1` rounding-boundary check) and at the DB layer (`chk_reward_pricing_settings_half_life`/`chk_reward_pricing_settings_multiplier` CHECK constraints on `reward_pricing_settings`). Since it's genuinely unreachable through the app's own write path, I'm leaving it unguarded — `decayDemand`'s guard exists because `elapsedSeconds` is *not* a validated input (it's wall-clock-derived and can legitimately be ≤0 from clock skew), which is a different situation. Adding a redundant guard here would go against this repo's "don't validate what can't happen" convention (`CLAUDE.md`). Fixed the two other review items (half-life rounding edge case, restored `index.test.ts` coverage) in 5883b53.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by ExpiredMinotaur** [comment]
@coderabbitai review

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by ExpiredMinotaur** [comment]
@coderabbitai review

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by ExpiredMinotaur** [comment]
@coderabbitai review

---
_Generated by [Claude Code](https://claude.ai/code)_
